### PR TITLE
[COM-1179] Add LinkIcon component and rename IconButton to ButtonIcon

### PR DESCRIPTION
**Pull request by alejandraarri**
**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** 
https://impulsumvc.atlassian.net/browse/COM-1179

**Does this PR introduce a breaking change?** 
No

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

File: packages/components/src/components/navigation/LinkIcon/LinkIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  if (!isValidElement(icon)) return null;
+
+  const iconElement = cloneElement(icon, { size } as Partial<IconProps>);
+  return isInline ? (
+    <chakra.span display="inline-flex" verticalAlign="middle" marginInline="0.25rem">
+      {iconElement}
+    </chakra.span>
+  ) : (
+    iconElement
+  );
```
Comment (guilean) [inline packages/components/src/components/navigation/LinkIcon/LinkIcon.tsx:36]: Is this logic needed for the `LinkIcon`?

File: packages/components/src/components/navigation/LinkIcon/LinkIcon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export default {
+  component: LinkIcon,
+  title: 'Components/Navigation/LinkIcon',
+  argTypes: {
+    variant: {
+      options: [...linkIconVariants],
+      control: { type: 'select' },
+    },
+    size: {
+      options: linkIconSizes,
```
Comment (guilean) [inline packages/components/src/components/navigation/LinkIcon/LinkIcon.stories.tsx:17]: Could we use `linkIconVariants` array directly? Like you did with `linkIconSizes`?

File: packages/components/src/theme/ComposerTheme/Components/Link.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const linkBaseStyle = {
   },
 };
 
-const generateLink = (colors: { default: string; hover: string; pressed: string; focus: string }) => {
+export const generateLink = (colors: { default: string; hover: string; pressed: string; focus: string }) => {
```
Comment (guilean) [inline packages/components/src/theme/ComposerTheme/Components/Link.ts:14]: Can we remove the `export`?

File: packages/components/src/theme/ComposerTheme/Components/Link.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ const getSizes = () => {
   return sizes;
 };
 
-const linkPrimary = generateLink({
+export const linkPrimary = generateLink({
```
Comment (guilean) [inline packages/components/src/theme/ComposerTheme/Components/Link.ts:70]: ditto

**Review by guilean — COMMENTED** [review]
Looking good! I've left some minor comments 🙏🏻

**Review by guilean — APPROVED** [review]

File: packages/components/src/components/form/IconButton/IconButton.tsx [hunk 5]
```diff
@@ -16,6 +16,8 @@ export const IconButton: FC<IconButtonProps> = forwardRef<IconButtonProps, 'butt
 
     const Icon = React.cloneElement(icon, { size: responsiveSize });
 
+    console.warn('Deprecated: The component "IconButton" is deprecated. Please use "ButtonIcon" instead.');
```
Comment (tkanzakic) [inline packages/components/src/components/form/IconButton/IconButton.tsx:19]: nit: can we add a @deprecated comment to the component so the editor warns if we try to use it and add a comment to the storybook too?

File: packages/components/src/components/form/IconButton/types.ts [hunk 6]
```diff
@@ -1,14 +1,15 @@
+import { ButtonIconSize, ButtonIconVariant, buttonIconSizes, buttonIconVariants } from '@components';
```
Comment (tkanzakic) [inline packages/components/src/components/form/IconButton/types.ts:1]: nit: can we merge this import with the one in line 5?

**Review by tkanzakic — APPROVED** [review]
couple of minor comments nad LGTM

**Pull request merged**